Index: monodevelop/configure.in
===================================================================
--- monodevelop.orig/configure.in	2012-05-15 17:16:56.000000000 +0100
+++ monodevelop/configure.in	2012-05-15 18:15:57.470072361 +0100
@@ -328,7 +328,6 @@
 src/addins/MonoDevelop.AssemblyBrowser/Makefile
 src/addins/MonoDevelop.SourceEditor2/Makefile
 src/addins/MonoDevelop.XmlEditor/Makefile
-src/addins/MonoDevelop.CodeMetrics/Makefile
 src/addins/MonoDevelop.Refactoring/Makefile
 src/addins/MonoDevelop.Debugger/Makefile
 src/addins/MonoDevelop.Debugger.Soft/Makefile
