Index: monodevelop/src/addins/NUnit/Makefile.am
===================================================================
--- monodevelop.orig/src/addins/NUnit/Makefile.am	2011-11-08 00:30:37.000000000 +0000
+++ monodevelop/src/addins/NUnit/Makefile.am	2011-11-08 00:48:51.000000000 +0000
@@ -13,10 +13,7 @@
 	$(GLIB_SHARP_LIBS) \
 	$(GTK_SHARP_LIBS) \
 	$(MONO_ADDINS_LIBS) \
-	-r:$(top_srcdir)/src/addins/NUnit/lib/nunit.core.dll \
-	-r:$(top_srcdir)/src/addins/NUnit/lib/nunit.core.interfaces.dll \
-	-r:$(top_srcdir)/src/addins/NUnit/lib/nunit.framework.dll \
-	-r:$(top_srcdir)/src/addins/NUnit/lib/nunit.util.dll \
+	$(shell pkg-config --libs nunit) \
 	-r:Mono.Cairo \
 	-r:Mono.Posix \
 	-r:System \
@@ -85,16 +82,7 @@
 	templates/NUnitTestClass.xft.xml
 
 
-NUNIT_FILES = \
-nunit.core.dll \
-nunit.framework.dll \
-nunit.core.interfaces.dll \
-nunit.util.dll
-
-SRC_NUNIT_FILES = $(addprefix $(srcdir)/lib/, $(NUNIT_FILES))
-BUILD_NUNIT_FILES = $(addprefix $(ADDIN_BUILD)/, $(NUNIT_FILES))
-
-all: $(ASSEMBLY) $(ASSEMBLY).mdb $(BUILD_NUNIT_FILES)
+all: $(ASSEMBLY) $(ASSEMBLY).mdb
 
 $(ADDIN_BUILD)/%: $(srcdir)/lib/%
 	mkdir -p $(ADDIN_BUILD)
@@ -107,9 +95,9 @@
 $(ASSEMBLY).mdb: $(ASSEMBLY)
 
 assemblydir = $(MD_ADDIN_DIR)/NUnit
-assembly_DATA = $(ASSEMBLY) $(ASSEMBLY).mdb $(SRC_NUNIT_FILES)
+assembly_DATA = $(ASSEMBLY) $(ASSEMBLY).mdb
 
-CLEANFILES = $(ASSEMBLY) $(ASSEMBLY).mdb $(BUILD_NUNIT_FILES)
-EXTRA_DIST = $(FILES) $(RES) $(SRC_NUNIT_FILES)
+CLEANFILES = $(ASSEMBLY) $(ASSEMBLY).mdb
+EXTRA_DIST = $(FILES) $(RES)
 
 include $(top_srcdir)/Makefile.include
Index: monodevelop/src/addins/NUnit/MonoDevelopNUnit.addin.xml
===================================================================
--- monodevelop.orig/src/addins/NUnit/MonoDevelopNUnit.addin.xml	2011-11-08 00:33:54.000000000 +0000
+++ monodevelop/src/addins/NUnit/MonoDevelopNUnit.addin.xml	2011-11-08 00:50:12.000000000 +0000
@@ -10,10 +10,6 @@
 
 	<Runtime>
 		<Import assembly="MonoDevelop.NUnit.dll" />
-		<Import assembly="nunit.core.dll" />
-		<Import assembly="nunit.core.interfaces.dll" />
-		<Import assembly="nunit.framework.dll" />
-		<Import assembly="nunit.util.dll" />
 	</Runtime>
 	
 	<Dependencies>
@@ -63,16 +59,6 @@
 
 	<!-- Extensions -->
 
-	<Extension path="/MonoDevelop/Core/SupportPackages">
-		<Condition id="PackageNotInstalled" name="mono-nunit" version="2.7">
-			<Package name="nunit" version="2.4" clrVersion="Default">
-				<Assembly file="nunit.core.dll" />
-				<Assembly file="nunit.core.interfaces.dll" />
-				<Assembly file="nunit.framework.dll" />
-			</Package>
-		</Condition>
-	</Extension>
-
 	<Extension path = "/MonoDevelop/Ide/Pads">
 		<SolutionPad id = "MonoDevelop.NUnit.TestPad" defaultLayout="*" defaultPlacement = "Left" defaultStatus="AutoHide" _label = "Unit Tests" icon = "nunit-pad-icon" class = "MonoDevelop.NUnit.TestPad">
 			<PadOption id = "ShowTestTime" _label = "Show Test Time" defaultValue = "False" />
Index: monodevelop/src/addins/NUnit/NUnitRunner/Makefile.am
===================================================================
--- monodevelop.orig/src/addins/NUnit/NUnitRunner/Makefile.am	2011-11-08 00:30:37.000000000 +0000
+++ monodevelop/src/addins/NUnit/NUnitRunner/Makefile.am	2011-11-08 00:48:51.000000000 +0000
@@ -1,9 +1,7 @@
 FILES = NUnitTestRunner.cs
 
 REFS =  \
-	-r:$(top_srcdir)/src/addins/NUnit/lib/nunit.core.dll \
-	-r:$(top_srcdir)/src/addins/NUnit/lib/nunit.core.interfaces.dll \
-	-r:$(top_srcdir)/src/addins/NUnit/lib/nunit.framework.dll \
+	$(shell pkg-config --libs nunit) \
 	-r:System \
 	-r:System.Xml
 
