Index: xamarin-pkg-monodevelop/src/addins/VersionControl/MonoDevelop.VersionControl.Git/build_libgit2.sh
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ xamarin-pkg-monodevelop/src/addins/VersionControl/MonoDevelop.VersionControl.Git/build_libgit2.sh	2015-08-27 10:56:12.353873460 +0100
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+pushd ../../../../external/libgit2/
+LIBGIT2SHA=`cat ../libgit-binary/libgit2_hash.txt`
+SHORTSHA=${LIBGIT2SHA:0:7}
+
+if [[ -d build ]]
+then
+    pushd build
+
+    if [[ -n $(ls libgit2-${SHORTSHA}.*) ]]
+    then
+        exit 0
+    else
+        popd
+        rm -rf build
+    fi
+fi
+
+mkdir build
+pushd build
+
+cmake -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo \
+      -DTHREADSAFE:BOOL=ON \
+      -DBUILD_CLAR:BOOL=OFF \
+      -DUSE_SSH=ON \
+      -DLIBGIT2_FILENAME=git2-$SHORTSHA \
+      -DCMAKE_OSX_ARCHITECTURES="i386;x86_64" \
+      -DCMAKE_SKIP_RPATH=TRUE \
+      ..
+
+cmake --build .
+popd
+popd
+
