Index: monodevelop/Makefile.am
===================================================================
--- monodevelop.orig/Makefile.am	2012-05-15 17:16:56.406504984 +0100
+++ monodevelop/Makefile.am	2012-05-15 18:24:11.825701399 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = ../external contrib src po man theme-icons build tests
+SUBDIRS = contrib src po man theme-icons build tests
 
 #capture aclocal flags for autoreconf
 ACLOCAL_AMFLAGS=$(ACLOCAL_FLAGS)
