Index: monodevelop-2.2/contrib/Mono.Cecil/Makefile.am
===================================================================
--- monodevelop-2.2.orig/contrib/Mono.Cecil/Makefile.am	2010-01-07 19:55:31.000000000 +0000
+++ monodevelop-2.2/contrib/Mono.Cecil/Makefile.am	2010-01-07 19:56:44.000000000 +0000
@@ -1,4 +1,5 @@
 BINDIR=$(top_builddir)/build/bin
+MCS=@CSC@
 
 all: $(CECIL)
 
@@ -11,7 +12,7 @@
 
 $(CECIL): $(CECIL_CSFILES)
 	mkdir -p $(BINDIR)
-	mcs $(CECIL_CSFILES) /debug /target:library /out:$(CECIL)
+	$(MCS) $(CECIL_CSFILES) /debug /target:library /out:$(CECIL)
 
 $(CECIL_DEBUG): $(CECIL)
 
@@ -24,7 +25,7 @@
 
 $(CECIL_MDB): $(CECIL_MDB_CSFILES) $(CECIL)
 	mkdir -p $(BINDIR)
-	mcs $(CECIL_MDB_CSFILES) /debug /target:library /r:$(CECIL) /d:CECIL /out:$(CECIL_MDB)
+	$(CSC) $(CECIL_MDB_CSFILES) /debug /target:library /r:$(CECIL) /d:CECIL /out:$(CECIL_MDB)
 
 $(CECIL_MDB_DEBUG): $(CECIL_MDB)
 
