Index: monodevelop-2.4+dfsg/src/addins/MonoDevelop.Debugger.Soft/Mono.Debugging.Soft/Makefile.am
===================================================================
--- monodevelop-2.4+dfsg.orig/src/addins/MonoDevelop.Debugger.Soft/Mono.Debugging.Soft/Makefile.am	2010-09-14 14:25:06.547939151 +0100
+++ monodevelop-2.4+dfsg/src/addins/MonoDevelop.Debugger.Soft/Mono.Debugging.Soft/Makefile.am	2010-09-14 14:25:13.467938607 +0100
@@ -32,7 +32,7 @@
 
 $(ASSEMBLY): $(build_sources) $(build_resources) $(DEPS) $(DBG_ASM)
 	mkdir -p $(ADDIN_BUILD)
-	$(CSC) $(CSC_FLAGS) -debug -out:$@ -keycontainer:$(SIGNING_KEY) -target:library $(REFS) $(build_deps) \
+	$(CSC) $(CSC_FLAGS) -debug -out:$@ -keyfile:$(SIGNING_KEY) -target:library $(REFS) $(build_deps) \
 		$(build_resources:%=/resource:%) $(build_sources)
 
 $(DBG_ASM): $(DBG_ASM_SRC)
Index: monodevelop-2.4+dfsg/src/core/Mono.Debugging/Makefile.am
===================================================================
--- monodevelop-2.4+dfsg.orig/src/core/Mono.Debugging/Makefile.am	2010-09-14 14:25:18.467939886 +0100
+++ monodevelop-2.4+dfsg/src/core/Mono.Debugging/Makefile.am	2010-09-14 14:25:28.530437511 +0100
@@ -82,7 +82,7 @@
 
 $(ASSEMBLY): $(build_sources) $(build_resources) $(SIGNING_KEY) $(DEPS)
 	mkdir -p $(ASSEMBLY_BUILD_DIR)
-	$(CSC) $(CSC_FLAGS) -debug -out:$@ /keycontainer:$(SIGNING_KEY) -target:library $(REFS) $(build_deps) $(build_resources:%=/resource:%) $(build_sources)
+	$(CSC) $(CSC_FLAGS) -debug -out:$@ /keyfile:$(SIGNING_KEY) -target:library $(REFS) $(build_deps) $(build_resources:%=/resource:%) $(build_sources)
 
 $(ASSEMBLY).mdb: $(ASSEMBLY)
 
