Index:  monodevelop-0.13.1/Core/src/MonoDevelop.Core/MonoDevelop.Core.Execution/ProcessService.cs
Description: Use x-terminal-emulator instead of xterm
Patch created by Mirco Bauer <meebey@meebey.net>
===================================================================
diff -urNad monodevelop-0.13.1~/Core/src/MonoDevelop.Core/MonoDevelop.Core.Execution/ProcessService.cs monodevelop-0.13.1/Core/src/MonoDevelop.Core/MonoDevelop.Core.Execution/ProcessService.cs
--- monodevelop-0.13.1~/Core/src/MonoDevelop.Core/MonoDevelop.Core.Execution/ProcessService.cs	2007-03-08 00:02:14.000000000 +0100
+++ monodevelop-0.13.1/Core/src/MonoDevelop.Core/MonoDevelop.Core.Execution/ProcessService.cs	2007-04-01 22:57:37.000000000 +0200
@@ -116,8 +116,8 @@
 				string additionalCommands = "";
 				if (!console.CloseOnDispose)
 					additionalCommands = @"echo; read -p 'Press any key to continue...' -n1;";
-				ProcessStartInfo psi = new ProcessStartInfo("xterm",
-					String.Format (@"-e ""cd {3} ; '{0}' {1} ; {2}""", command, arguments, additionalCommands, workingDirectory.Replace (" ", "\\ ")));
+				ProcessStartInfo psi = new ProcessStartInfo("x-terminal-emulator",
+					String.Format (@"-e /bin/sh -c ""cd {3} ; '{0}' {1} ; {2}""", command, arguments, additionalCommands, workingDirectory.Replace (" ", "\\ ")));
 				psi.UseShellExecute = false;
 				
 				if (workingDirectory != null)
